pkg/model/dev.go

```diff
@@ -331,6 +334,24 @@ func (dev *Dev) setDefaults() error {
 	return nil
 }
 
+func (dev *Dev) setRunAsUserDefaults(main *Dev) {
+	if !main.PersistentVolumeEnabled() {
+		return
+	}
+	if dev.SecurityContext == nil {
+		dev.SecurityContext = &SecurityContext{}
+	}
+	if dev.SecurityContext.RunAsUser == nil {
```

Comment:
what's the purpose of this? I'd rather not set something that the user is not setting herself, to limit the stuff we change.

Reply:
it is one of the main things this issue is about. Currently, we document that root images work by default. if the user wants to use a non root dev image, `runAsUser` must be defined.
But the current behavior is broken, if `runAsUser` is set in the k8 manifest, a root dev image does not work. several users have been reported this issue